### DISCUSSION TITLE
Encodings for each value type

### DISCUSSION THREAD

**Discussion by parsisoftware** [pyg-team/pytorch_geometric#8342] [Q&A]
Hello,
I have created my graphs using the following link:
https://pytorch-geometric.readthedocs.io/en/latest/tutorial/load_csv.html

I want to know the different ways of encoding different value types into the numerical feature representation.
For example we can encode the string type using a SequenceEncoder or categorical labels, considering the values exists in the dataset.
I want to have all possible ways of encoding String, float or boolean types, etc.
Please present a link or guide me if possible.

Thank you

**Comment by rusty1s** [comment]
You might find https://github.com/pyg-team/pytorch-frame useful :)

**Comment by parsisoftware** [comment]
Thank you rusty,
Can I use the encoders as the following:

```python
from torch_frame import stype
from torch_frame.nn import (
EmbeddingEncoder,
FTTransformer,
LinearEmbeddingEncoder,
LinearEncoder,
)

movie_x, movie_mapping = load_node_csv(
movie_path, index_col='movieId', encoders={
'title': LinearEmbeddingEncoder(),
'genres': EmbeddingEncoder()
})
```

And, Is there any documentation that explain the application of all these encoders?

**Comment by parsisoftware** [comment]
Would you please answer my question??

**Reply by akihironitta** [reply]
Hi @parsisoftware, https://github.com/pyg-team/pytorch-frame is still at an early stage where the docs might not be sufficiently descriptive. For now, I'd suggest looking at examples (and the codebase if needed). If you have any question about PyTorch Frame, feel free to submit a GitHub issue to the repository!
